csvcubed/tests/unit/utils/qb/test_component.py

```diff
@@ -0,0 +1,80 @@
+from pathlib import Path
+import pytest
+
+from csvcubed.utils.qb.components import (
+    ComponentPropertyType,
+    get_printable_component_property,
+    get_printable_component_property_type,
+)
+
+
+def test_printable_component_property_url():
+    input_file_path = Path("folder/sub-folder/file.csv")
+    component_property = "http://purl.org/linked-data/sdmx/2009/attribute#obsStatus"
+    printable_component_property = get_printable_component_property(
+        input_file_path, component_property
+    )
+    assert printable_component_property == "http://purl.org/linked-data/sdmx/2009/attribute#obsStatus"
+
+
+def test_printable_component_property_supported_file_path():
+    input_file_path = Path("folder/sub-folder/file.csv")
+    component_property = (
+        "file://folder/sub-folder/sub-sub-folder/other-file.csv#property-type/type"
+    )
+    printable_component_property = get_printable_component_property(
+        input_file_path, component_property
+    )
+    assert printable_component_property == "./sub-sub-folder/other-file.csv#property-type/type"
+
+
+def test_printable_component_property_unsupported_file_path():
+    input_file_path = Path("folder/sub-folder/file.csv")
+    component_property = "file://folder/new-folder/sub-folder/sub-sub-folder/other-file.csv#property-type/type"
+    printable_component_property = get_printable_component_property(
+        input_file_path, component_property
+    )
+    assert (
+        printable_component_property
+        == "folder/new-folder/sub-folder/sub-sub-folder/other-file.csv#property-type/type"
+    )
+
+
+def test_printable_component_property_type_dimension():
+    """
+    Should return "Dimension" when the input is http://purl.org/linked-data/cube#DimensionProperty
+    """
+    property_type = get_printable_component_property_type(
+        "http://purl.org/linked-data/cube#DimensionProperty"
+    )
+    assert property_type == ComponentPropertyType.Dimension.value
+
+
+def test_printable_component_property_type_attribute():
+    """
+    Should return "Attribute" when the input is http://purl.org/linked-data/cube#AttributeProperty
+    """
+    property_type = get_printable_component_property_type(
+        "http://purl.org/linked-data/cube#AttributeProperty"
+    )
+    assert property_type == ComponentPropertyType.Attribute.value
+
+
+def test_printable_component_property_type_measure():
+    """
+    Should return "Attribute" when the input is http://purl.org/linked-data/cube#MeasureProperty
```

Comment:
I agree with the behaviour but not the comment.

Reply:
This is now resolved.